scripts/convert_msnet.py

```
@@ -0,0 +1,1179 @@
+        run_list = ", ".join(f"'{r.replace(chr(39), chr(39) + chr(39))}'" for r in run_file_names)
+        sql = (
+            f"SELECT count(*) AS total, "
+            f"count_if(abs(observed_mz - calculated_mz) > {qc_limit}) AS outliers "
+            f"FROM parquet_scan('{safe_path}/**/*.parquet', hive_partitioning=true) "
+            f"WHERE organism = '{organism.replace(chr(39), chr(39) + chr(39))}' "
+            f"AND run_file_name IN ({run_list})"
+        )
+        row = con.execute(sql).fetchone()
+        total = row[0] or 0
+        outliers = row[1] or 0
+    except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)